src/python/torchdistx/optimizers/anyprecision_optimizer.py

```diff
@@ -72,6 +117,31 @@ def __init__(
 
         super().__init__(params, defaults)
 
+        # confirm bfloat16 support if applicable
+        if (
+            torch.bfloat16
+            in [
+                momentum_dtype,
+                variance_dtype,
+            ]
+            or torch.bfloat16 in [compensation_buffer_dtype]
+            and use_kahan_summation
+        ):
+            gpu_support, network_support = self.verify_bfloat_support()
+
+            if not gpu_support or not network_support:
+                if not gpu_support:
```

Comment:
Let's just include these print()s in the error msg. Something like:

```
reason = ""
if not gpu_support:
    reason += "GPU does not support"
if not network_support:
    reason += "NCCL does not support"

raise ValueError(f"Missing bf16 support: {reason}")
```

Reply:
that's much cleaner.  Updated.